### DISCUSSION TITLE
Custom Constant Buffers

### DISCUSSION THREAD

**Discussion by jeanlemotan** [google/filament#4418] [Q&A]
Is there any way to use extra, custom constant buffers? 
I need to be able to set extra scene data that is consumed in materials without changing every material from every object in the scene.
This can only be done with textures at the moment but this wastes a sampler and is somewhat cumbersome to pack arbitrary data in one or multiple texture.

**Comment by romainguy** [comment]
This is not possible at the moment and doing so would require somehow passing that information to the materials when they are generated/compiled which could be an issue. Do you need a lot of data? We could add the ability to pass a few user `float4` as frame data.

**Reply by jeanlemotan** [reply]
You're right, the constant buffers would need to be declared ahead of time so that the materials contain their definition.
A few float4 would definitely help for stuff like scripted eye adaptation, debugging values etc.

**Reply by romainguy** [reply]
@pixelflinger What do you think?
